### PR TITLE
Added nn.ModuleFromCriterion to wrap a called Criterion.

### PR DESCRIPTION
**Pull request by fidlej**
@wojciechz @koraykv
I took an inspiration from Wojciech.
And I provided the support for Criterions inside of a nngraph.
A Criterion is automatically wrapped by a module,
when converting the criterion to a node.

Is this support matching your needs?

### CODE REVIEW

**Comment by Atcold** [comment]
@fidlej, could you please add some documentation?

**Comment by fidlej** [comment]
What is unclear?
I do not recommend using Criterions in a graph. The gradient with respect to the target will be zero.

**Comment by Atcold** [comment]
All other features are documented. It would be useful to have a working example showing how it can be used.
Why are you saying that the gradient will be zero?
